tests/io/hdf5_test.py

```diff
@@ -279,6 +279,18 @@ def test_data_group():
             'data_array': array_2d,
             'dataset': sc.Dataset({'a': array_1d, 'b': array_2d}),
             'data group': sc.DataGroup({'v.same': vector, 'm/copy': matrix.copy()}),
+            'integer': 513,
+            'numpy int64': np.int64(9),
+            'numpy int32': np.int32(-23),
+            'numpy uint64': np.uint64(9312),
+            'numpy uint32': np.uint32(66134),
+            'float': 7.81,
+            'numpy float64': np.float64(5235.1),
+            'numpy float32': np.float32(0.031),
+            'boolean': True,
+            'numpy bool_': np.bool_(False),
+            'string': 'a string for testing',
```

Comment:
I have seen NeXus files that use bytesrings, can we support that?

Reply:
What do you mean by bytestring? Python's `bytes`? h5py seems to store `str` as `bytes`. This is why the loader calls `decode`.

Reply:
Yes, `bytes`, iirc.

Reply:
Done